### PR TITLE
SUSE.Linux.Events.Crontab: Add task execution artifacts

### PR DESCRIPTION
**Pull request by xTeixeira**
Parse cron task executions from syslog and journal.

### CODE REVIEW

File: artifacts/definitions/SUSE/Linux/Events/Cron.yaml [hunk 1]
```diff
@@ -12,3 +26,31 @@ sources:
     query: |
       SELECT * from cronsnoop(spool_dir='/var/spool/cron/tabs',system_dirs=['/etc/crontab', '/etc/cron.d', '/etc/cron.hourly', '/etc/cron.monthly', '/etc/cron.daily', '/etc/cron.weekly'])
 
+  - name: JournalTaskExecs
+    precondition: SELECT OS From info() where OS = 'linux'
+    description: Collect cron task executions from systemd journal
+    query: |
+      LET cron_exec = SELECT REALTIME_TIMESTAMP, _PID, MESSAGE, _SYSTEMD_UNIT, grok(grok=CronGrok, data=MESSAGE) AS Event
+        FROM watch_journal()
+        WHERE _TRANSPORT != 'kernel' AND _SYSTEMD_UNIT = CronSystemdUnit and Event.action = "CMD"
```
Comment (djoreilly) [inline artifacts/definitions/SUSE/Linux/Events/Cron.yaml:35]: This is not working for me on Tumbleweed - I have the patch for https://github.com/SUSE/linux-security-sensor/pull/75 installed.  
```
 # crontab -l
*/1 * * * * /usr/bin/date
```
 Looking at the JSON from `watch_journal()` it seems `_SYSTEMD_UNIT` is `session-c387.scope` and not `cron.service`.

**Review by djoreilly — CHANGES_REQUESTED** [review]

File: artifacts/definitions/SUSE/Linux/Events/Cron.yaml (hunk 1 above)
Comment (xTeixeira) [reply artifacts/definitions/SUSE/Linux/Events/Cron.yaml:35]: It looks like Tumbleweed uses a custom PAM config for cronie which causes it to create a new session scope for executing cron tasks, meaning the logs become unassociated with the original systemd unit.

I'll see if I can modify this to not rely on `_SYSTEMD_UNIT` to query events, as it won't always be the same. Relying on `_COMM = "cron"` might be enough.

Comment (xTeixeira) [reply artifacts/definitions/SUSE/Linux/Events/Cron.yaml:35]: Opened PR #80 against 0.7.0 with this problem fixed.

Comment (xTeixeira) [reply artifacts/definitions/SUSE/Linux/Events/Cron.yaml:35]: @djoreilly Do we still want this merged on 0.6.7 or are we only going to work on 0.7.0 going forward?

Comment (djoreilly) [reply artifacts/definitions/SUSE/Linux/Events/Cron.yaml:35]: I think we are only going to work on 0.7.0 from now on

**Comment by djoreilly** [comment]
moved to https://github.com/SUSE/linux-security-sensor/pull/80

**Pull request closed without merging**